Added package for Paradiseo.

jppelteret opened the pull request:
This package mirrors the Homebrew/Linuxbrew package found [here](https://github.com/jppelteret/homebrew-science-extras/blob/master/paradiseo.rb). Unfortunately the source repository and packages are in a bit of a state, so the use of a specific git revision is recommended. Patches have been applied to make it compile on both OS X and Linux.

mathstuf commented on var/spack/repos/builtin/packages/paradiseo/package.py:
```diff
@@ -0,0 +1,71 @@
+
+        dsuf = 'dylib' if sys.platform == 'darwin' else 'so'
+        options.extend([
+            '-DCMAKE_BUILD_TYPE:STRING=%s' % ('DEBUG' if '+debug' in spec else 'RELEASE'),
```
Usually, these are `Debug` and `Release` rather than all-cap, but if this is what the project expects, that's fine.

mathstuf commented on var/spack/repos/builtin/packages/paradiseo/package.py:
```diff
@@ -0,0 +1,71 @@
+        options = []
+        options.extend(std_cmake_args)
+
+        dsuf = 'dylib' if sys.platform == 'darwin' else 'so'
```
Unused variable?

mathstuf commented on var/spack/repos/builtin/packages/paradiseo/package.py:
```diff
@@ -0,0 +1,71 @@
+        if '+mpi' in spec:
+            options.extend([
+                '-DMPI:BOOL=TRUE',
+                '-DCMAKE_C_COMPILER=%s' % join_path(self.spec['mpi'].prefix.bin, 'mpicc'), # FIXME: avoid hardcoding mpi wrappers names
```
CMake's `FindMPI` module usually handles this fine; does it really need to use the wrappers directly?

jppelteret commented:
Thanks for the suggestions, @mathstuf . I've addressed the points that you've made. Indeed, the MPI wrappers were found through CMake's `find_package`.

mathstuf commented on var/spack/repos/builtin/packages/paradiseo/package.py:
```diff
@@ -38,28 +39,14 @@ def install(self, spec, prefix):
-        # MPI
-        if '+mpi' in spec:
-            options.extend([
-                '-DMPI:BOOL=TRUE',
```
Hmm. This one might still be important, but can probably move up with `SMP` and `EDO` above.

jppelteret replied:
Yes, of course. I had meant to do that :-/ I'll fix it tomorrow and squash all of the commits.

tgamblin commented:
Thanks!
